src/server/request_handler/default_request_handler.ts

```diff
@@ -566,4 +577,25 @@ export class DefaultRequestHandler implements A2ARequestHandler {
         // Send push notification in the background.
         this.pushNotificationSender?.send(task);
     }
+
+    // Check if the params for the TasksList function are valid
+    private paramsTasksListAreValid(params: ListTasksParams): boolean {
+        if(params.pageSize !== undefined && (params.pageSize > 100 || params.pageSize < 1)) {
+            return false;
+        }
+        if(params.pageToken !== undefined && Buffer.from(params.pageToken, 'base64').toString('base64') !== params.pageToken){
+            return false;
+        }
+        if(params.historyLength !== undefined && params.historyLength<0){
+            return false;
+        }
+        if(params.lastUpdatedAfter !== undefined && !isValidUnixTimestampMs(params.lastUpdatedAfter)){
+            return false;
+        }
+        const terminalStates: string[] = ["completed", "failed", "canceled", "rejected"];
```

Comment:
Is there a particular reason why only the terminal states are available for querying? The spec indicates that all task states are available for query.